__main__.py

```diff
@@ -3,81 +3,149 @@
 import datetime as dt
 import json
 import sys
+from enum import Enum
 from pathlib import Path
-from typing import NoReturn
+from typing import TYPE_CHECKING, Any, ClassVar, NewType, NoReturn, TypedDict, cast
 
 import requests
 from dahlia import dahlia, dprint
+from requests.exceptions import HTTPError
+
+if TYPE_CHECKING:
+    from typing_extensions import Self
+
+
+__version__ = "0.1.0"
+
+
+class Status(str, Enum):
+    ACCEPTED = "Accepted"
+    ACTIVE = "Active"
+    DEFERRED = "Deferred"
+    DRAFT = "Draft"
+    FINAL = "Final"
+    PROVISIONAL = "Provisional"
+    REJECTED = "Rejected"
+    REPLACED = "Replaced"
+    WITHDRAWN = "Withdrawn"
+    SUPERSEDED = "Superseded"
```

Comment:
Using an `Enum` to store possible PEP statuses. The values match what's returned by their API, so it's possible to do `Status("Accepted")` to get the `Status.ACCESSED` instance, for example. It also inherits from `str`, so they can still be used where a `str` would be expected - mostly.

Reply:
perhaps `auto()` and use `Status.<Member>.name.title()` instead?

Reply:
> perhaps `auto()` and use `Status.<Member>.name.title()` instead?

IMO it's better to do it explicitly like this. Using `name.title()` would obfuscate the connection between the names in the enum and the names in PEP's API.

Reply:
> > perhaps `auto()` and use `Status.<Member>.name.title()` instead?
> 
> IMO it's better to do it explicitly like this. Using `name.title()` would obfuscate the connection between the names in the enum and the names in PEP's API.

How so?

Reply:
> > > perhaps `auto()` and use `Status.<Member>.name.title()` instead?
> > 
> > 
> > IMO it's better to do it explicitly like this. Using `name.title()` would obfuscate the connection between the names in the enum and the names in PEP's API.
> 
> How so?

Well, for the currently available values, this might work, but this correlation isn't guaranteed by anything I believe. If, for example, one of the values was "In Revision", how would one get to that from `IN_REVISION`? Additional logic would be needed.

IMO it's better to explicitly assign the values to the corresponding names. This might make it easier to read (one can clearly see how the values from the API are being turned into members of the enum) and more future-proof (in case they ever add a new value that doesn't follow the made-up rule of "simply convert from all caps to title case").

Reply:
Using `auto()` is safer because it eliminates the risk of making a typo like `SUPERSEDED = "Superseeded"`. Also DRY :)